Bug: baidu engine

**Version of SearXNG, commit number if you are using on master branch and stipulate if you forked SearXNG**
Repository: https://github.com/searxng/searxng
Branch: master
Version: 2025.4.7+b146b745a
<!-- Check if these values are correct -->

**How did you install SearXNG?**
<!-- Did you install SearXNG using the official wiki or using searxng-docker
or manually by executing the searx/webapp.py file? -->
**What happened?**
<!-- A clear and concise description of what the bug is. -->

**How To Reproduce**
<!-- How can we reproduce this issue? (as minimally and as precisely as possible) -->

**Expected behavior**
<!-- A clear and concise description of what you expected to happen. -->

**Screenshots & Logs**
<!-- If applicable, add screenshots, logs to help explain your problem. -->

**Additional context**
<!-- Add any other context about the problem here. -->

**Technical report**

Error
  * Error: 1 redirects, maximum: 0
  * Percentage: 100
  * Parameters: `('200', 'OK', 'wappass.baidu.com')`
  * File name: `searx/search/processors/online.py:125`
  * Function: `_send_http_request`
  * Code: `count_error(`

Error
  * Error: json.decoder.JSONDecodeError
  * Percentage: 100
  * Parameters: `('Expecting value',)`
  * File name: `searx/engines/baidu.py:101`
  * Function: `response`
  * Code: `data = json.loads(text, strict=False)`




](is:issue Bug: baidu Error: 1 redirects, maximum: 0 Parameters: ('200', 'OK', 'wappass.baidu.com') File name: searx/search/processors/online.py:125 Error Function: _send_http_request Code: count_error( Error: json.decoder.JSONDecodeError Parameters: ('Expecting value',) File name: searx/engines/baidu.py:101 Error Function: response Code: data = json.loads(text, strict=False))](is:issue Bug: baidu Error: 1 redirects, maximum: 0 Parameters: ('200', 'OK', 'wappass.baidu.com') File name: searx/search/processors/online.py:125 Error Function: _send_http_request Code: count_error( Error: json.decoder.JSONDecodeError Parameters: ('Expecting value',) File name: searx/engines/baidu.py:101 Error Function: response Code: data = json.loads(text, strict=False))


How can I use baidu？

--- a/searx/engines/baidu.py
+++ b/searx/engines/baidu.py
@@ -89,12 +89,13 @@ def request(query, params):
             query_params["paramList"] += f",timestamp_range={past}-{now}"
 
     params["url"] = f"{query_url}?{urlencode(query_params)}"
+    params["allow_redirects"] = False
     return params
 
 
 def response(resp):
     # Detect Baidu Captcha, it will redirect to wappass.baidu.com
-    if 'wappass.baidu.com' in str(resp.url):
+    if 'wappass.baidu.com/static/captcha' in resp.headers.get('Location', ''):
         raise SearxEngineCaptchaException()
 
     text = resp.text

--- a/searx/engines/baidu.py
+++ b/searx/engines/baidu.py
@@ -13,7 +13,7 @@ from html import unescape
 import time
 import json
 
-from searx.exceptions import SearxEngineAPIException
+from searx.exceptions import SearxEngineAPIException, SearxEngineCaptchaException
 from searx.utils import html_to_text
 
 about = {
@@ -93,6 +93,9 @@ def request(query, params):
 
 
 def response(resp):
+    # Detect Baidu Captcha, it will redirect to wappass.baidu.com
+    if 'wappass.baidu.com' in str(resp.url):
+        raise SearxEngineCaptchaException()
 
     text = resp.text
     if baidu_category == 'images':

